chips/nrf5x/src/gpio.rs

```diff
@@ -504,26 +506,41 @@ impl<'a> hil::gpio::Interrupt<'a> for GPIOPin<'a> {
     }
 
     fn enable_interrupts(&self, mode: hil::gpio::InterruptEdge) {
-        if let Ok(channel) = self.allocate_channel() {
-            let polarity = match mode {
-                hil::gpio::InterruptEdge::EitherEdge => Config::POLARITY::Toggle,
-                hil::gpio::InterruptEdge::RisingEdge => Config::POLARITY::LoToHi,
-                hil::gpio::InterruptEdge::FallingEdge => Config::POLARITY::HiToLo,
-            };
-            let pin: u32 = (GPIO_PER_PORT as u32 * self.port as u32) + self.pin as u32;
-            self.gpiote_registers.config[channel]
-                .write(Config::MODE::Event + Config::PSEL.val(pin) + polarity);
-            self.gpiote_registers.intenset.set(1 << channel);
+        let channel = if let Some(chan) = self.allocated_channel.get() {
+            // We only support one interrupt mode per pin, despite the
+            // hardware supporting multiple. This is to comply with
+            // expectations in other Tock components, such as the button
+            // driver which re-registers interrupts for a restarted app,
+            // assuming the old ones to be overwritten.
+            chan
+        } else if let Ok(chan) = self.allocate_channel() {
+            // Don't have a channel yet, got a new one:
+            chan
         } else {
             debug!("No available GPIOTE interrupt channels");
-        }
+            return;
+        };
+
+        // Remember that we have allocated this channel for this pin:
+        self.allocated_channel.set(channel);
+
+        let polarity = match mode {
+            hil::gpio::InterruptEdge::EitherEdge => Config::POLARITY::Toggle,
+            hil::gpio::InterruptEdge::RisingEdge => Config::POLARITY::LoToHi,
+            hil::gpio::InterruptEdge::FallingEdge => Config::POLARITY::HiToLo,
+        };
+        let pin: u32 = (GPIO_PER_PORT as u32 * self.port as u32) + self.pin as u32;
+        self.gpiote_registers.config[channel]
+            .write(Config::MODE::Event + Config::PSEL.val(pin) + polarity);
+        self.gpiote_registers.intenset.set(1 << channel);
     }
 
     fn disable_interrupts(&self) {
-        if let Ok(channel) = self.find_channel(self.pin) {
+        if let Some(channel) = self.allocated_channel.get() {
             self.gpiote_registers.config[channel]
                 .write(Config::MODE::CLEAR + Config::PSEL::CLEAR + Config::POLARITY::CLEAR);
             self.gpiote_registers.intenclr.set(1 << channel);
+	    self.allocated_channel.take();
```

Comment:
Change to `.clear()` to make the intent more obvious?

Reply:
Changed and fixed formatting.